# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: 58f77fa35eea
# Full Hash: 58f77fa35eeab8de03c52f316d9f0553685a5f8d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-03-15 03:42:43
# Regressor Bug: 1532661
# File Overlap Count: 2
# Description:
#   Bug 1532661 - Part 3: Add WillSetX and DidSetX callbacks for BrowsingContext synced fields, r=farre
#   
#   Depends on D22191
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22192
# ==============================================================================

diff -r a39da23d450b -r 58f77fa35eea docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Thu Mar 14 18:51:05 2019 +0000
+++ b/docshell/base/BrowsingContext.cpp	Thu Mar 14 18:51:07 2019 +0000
@@ -688,7 +688,9 @@
                                          ContentParent* aSource) {
 #define MOZ_BC_FIELD(name, ...)                         \
   if (m##name) {                                        \
+    aBrowsingContext->WillSet##name(*m##name, aSource); \
     aBrowsingContext->m##name = std::move(*m##name);    \
+    aBrowsingContext->DidSet##name(aSource);            \
     m##name.reset();                                    \
   }
 #include "mozilla/dom/BrowsingContextFieldList.h"